draft-ietf-ppm-dap.md

```diff
@@ -1844,24 +1813,20 @@ Continuing, the Leader proceeds as follows with each report:
    If `outbound != None`, then the Leader stores `state` and `outbound` and
    proceeds to {{aggregation-leader-continuation}}. If `outbound == None`, then
    the preparation process is complete: either `state == Rejected()`, in which
-   case the Leader rejects the report and removes it from the candidate set; or
+   case the Leader rejects the report and removes it from the candidate set or
    `state == Finished(out_share)`, in which case preparation is complete and the
    Leader updates the relevant batch bucket with `out_share` as described in
    {{batch-buckets}}.
 
 1. Else if the type is "reject", then the Leader rejects the report and removes
    it from the candidate set. The Leader MUST NOT include the report in a
-   subsequent aggregation job, unless the error is `report_too_early`, in which
-   case the Leader MAY include the report in a subsequent aggregation job.
+   subsequent aggregation job, unless the error is `report_too_early`.
 
 1. Else the type is invalid, in which case the Leader MUST abort the
    aggregation job.
 
-When the Leader updates a batch bucket based on `out_share`, it MUST also store
-the report ID for replay protection.
```

Comment:
Prescribe what, not how: we don't need to tell implementations when to store things.

Reply:
In c717841, I re-introduced discussion of storing report IDs. In this specific case, I moved it to L1826, where we are discussing the case where the Leader recovers an `out_share`. I did remove a MUST, because while discussing storing the report ID is helpful for clarity, the MUST here is to prevent replays, not any particular storage strategy.